plugins/cilium-cni/cmd/endpoint.go

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/api/v1/models"
+	ipamOption "github.com/cilium/cilium/pkg/ipam/option"
+	"github.com/cilium/cilium/plugins/cilium-cni/types"
+)
+
+// EndpointConfigurator returns a list of endpoint configurations for a given
+// CNI ADD invocation. If the CNI ADD invocation should result in multiple endpoints
+// being created, it may return multiple endpoint configurations, one for each endpoint.
+type EndpointConfigurator interface {
+	GetConfiguration(p ConfigurationParams) ([]EndpointConfiguration, error)
```

Comment:
I'd probably rename this to plural `GetConfigurations`, even though in the default case it will only return a single default configuration.

```suggestion
	GetConfigurations(p ConfigurationParams) ([]EndpointConfiguration, error)
```